htdocs/core/actions_addupdatedelete.inc.php

```diff
@@ -74,7 +74,9 @@
 		} elseif (preg_match('/^(integer|price|real|double)/', $object->fields[$key]['type'])) {
 			$value = price2num(GETPOST($key, 'alphanohtml')); // To fix decimal separator according to lang setup
 		} elseif ($object->fields[$key]['type'] == 'boolean') {
-			$value = ((GETPOST($key) == '1' || GETPOST($key) == 'on') ? 1 : 0);
+			$value = (GETPOST($key) == '1' ? 1 : 0);
+		} elseif ($object->fields[$key]['type'] == 'reference') {
+			$value = array_keys($object->param_list)[GETPOST($key)].','.GETPOST($key.'2');
```

Comment:
I am not sure this syntax is compatible with PHP 5.6. Can you check ? If not, just store array_keys into a temporary var $tmparraykey